### DISCUSSION TITLE
How to uninstall poetry

### DISCUSSION THREAD

**Discussion by JustMyD** [python-poetry/poetry#9568] [Q&A] [answered]
Hello guys.
Previously i have installed poetry by curl "curl -sSL https://install.python-poetry.org | python3 -". Today i wanted to delete poetry and reinstall, so i just remove 'poetry' file from '/home/user/.local/bin'.  Another command "curl -sSL https://install.python-poetry.org | python3 -" tels me that poetry already installed.
Is there any way to completely remove all poetry files, or maybe i can restore file in '/home/user/.local/bin'?
P.s. i found poetry in pacman, so i just wondering if it's possible to find and delete old poetry files (installed with curl)

**Comment by Secrus** [comment, marked as answer]
You might want to remove `/home/user/.local/share/pypoetry`, since that is where the app lives. The file from `.local/bin` is just a launcher

**Reply by JustMyD** [reply]
thank you!

**Reply by ManiMatter** [reply]
This worked for me:
`curl -sSL https://install.python-poetry.org | python3 - --uninstall`

**Comment by nehubuser** [comment]
1.  curl -sSL https://install.python-poetry.org | python3 - --uninstall - didn't work. 
2. Removing /home/user/.local/share/pypoetry also didn't work. 

It still exist in my system. 

HELP. How do I remove it?

**Reply by Pei2tech** [reply]
you may check $HOME/.poetry and remove it.

**Comment by MukundaKatta** [comment]
@nehubuser if `--uninstall` didn't clear it, you probably have multiple parallel installs (curl-install, pipx, pacman/apt, pip are all independent). Remove each: `rm -rf ~/.local/share/pypoetry`, `rm -f ~/.local/bin/poetry`, `pipx uninstall poetry`, `sudo pacman -Rs poetry` (or equivalent for your distro), and `python3 -m pip uninstall poetry`. Then `which -a poetry` tells you if any install survived, and `hash -r` flushes your shell's cached path so the old binary stops showing up.
